src/wallet/test/silentpayment_tests.cpp

```diff
@@ -0,0 +1,161 @@
+#include <wallet/silentpayments.h>
+#include <test/data/bip352_send_and_receive_vectors.json.h>
+
+#include <test/util/setup_common.h>
+#include <hash.h>
+
+#include <boost/test/unit_test.hpp>
+#include <test/util/json.h>
+#include <vector>
+#include <util/bip32.h>
+#include <wallet/wallet.h>
+
+namespace wallet {
+BOOST_FIXTURE_TEST_SUITE(silentpayment_tests, BasicTestingSetup)
+
+CKey ParseHexToCKey(std::string hex) {
+    CKey output;
+    auto hex_data = ParseHex(hex);
+    output.Set(hex_data.begin(), hex_data.end(), true);
+    return output;
+};
+
+CKey GetKeyFromBIP32Path(std::vector<std::byte> seed, std::vector<uint32_t> path)
+{
+    CExtKey key_parent, key_child;
+    key_parent.SetSeed(seed);
+    for (auto index : path) {
+        BOOST_CHECK(key_parent.Derive(key_child, index));
+        std::swap(key_parent, key_child);
+    }
+    return key_parent.key;
```

Comment:
In 78d1426fabd7b5b6cdd28de508a1692fce291174 "Add BIP352 test vectors as unit tests"

No need to have two `CExtKey`s and swapping them, you can just derive a key into itself.

```suggestion
    CExtKey key;
    key.SetSeed(seed);
    for (auto index : path) {
        BOOST_CHECK(key.Derive(key, index));
    }
    return key.key;
```

Reply:
Done